Avoid forcing mangohud to run exec lines when possible

GloriousEggroll opened the pull request:
Mainly, this avoids hitting this bug from Goverlay's options
except for when the user defines a custom script to execute:
https://github.com/flightlessmango/MangoHud/issues/80

Also it converts several lines into a few single lines,
avoids needing to write distro/version/kernel/session to extra files,
and is generally cleaner/easier to read.

![Screenshot from 2022-08-06 23-48-42](https://user-images.githubusercontent.com/11287837/183277289-e7f9e406-dd38-47b6-a226-b52dccd507c5.png)

benjamimgois commented:
Hey @TheEvilSkeleton thankyou very much for this changes. They are indeed much better. I'll merge it right now.

TheEvilSkeleton commented:
No problem ^^

TheEvilSkeleton commented:
@benjamimgois also forgot to mention, I also made created an issue in the MangoHud repository, here: https://github.com/flightlessmango/MangoHud/issues/115

benjamimgois commented:
Thanks for the help man ! Actually the heavy leafting is done by MangoHud, i'm just trying to make it easier to everyone use. Be my guest to any other suggestions.

resynth1943 commented:
Looks good, @TheEvilSkeleton. Thank you for your contribution. :heart:

GloriousEggroll committed:
Avoid forcing mangohud to run exec lines when possible

Mainly, this avoids hitting this bug from Goverlay's options
except for when the user defines a custom script to execute:
https://github.com/flightlessmango/MangoHud/issues/80

Also it converts several lines into a few single lines,
avoids needing to write distro/version/kernel/session to extra files,
and is generally cleaner/easier to read.

Fijxu commented:
Tested it myself, it works good without problems ;)
The only downside is that when the kernel is updated, the string where the kernel information is displayed does not change, but this is not a big deal, it is something very minor

Fijxu commented:
This change is not longer necessary (i think) because it's already fixed from the mangohud side, i can launch apex legends without any problems using the `exec` lines provided by Goverlay

https://github.com/flightlessmango/MangoHud/commit/a05dd84bf521aaf222fd9d3f0550165a07a10f15

![](https://ii.zzls.xyz/ODGOTNI.png)

GloriousEggroll commented:
If we want to keep the exec lines they can still be converted to a single line I think without writing the output to a separate file

GloriousEggroll commented:
closed in favor of https://github.com/benjamimgois/goverlay/pull/13
